### PR TITLE
docs: clarify the use of classes and interfaces in style guide

### LINKED ISSUE #19632: Why styleguide suggest use class instead of interface?
In style guid, it says consider use class instead of interface
https://angular.io/guide/styleguide#interfaces

Also in the example file hero.model.ts it's using class:
export class Hero {
  id: number;
  name: string;
}

I think in this case better use interface instead of class? As typescript best practice is that in this case. Unlike classes, interfaces are completely removed during compilation and so they will not add any unnecessary bloat to our final JavaScript code.

Any reason to use class instead of interface in this case?? Or it's better to use interface in this case?

### PR DESCRIPTION
**Pull request by Toxicable**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Documentation content changes
```
A follow up from our discussion about #19632

closes #19632

### CODE REVIEW

**Comment by kapunahelewong** [comment]
LGTM. Thank you @Toxicable 

@gkalpak Ready for your review.

File: aio/content/guide/styleguide.md [hunk 1]
```diff
@@ -1904,7 +1904,18 @@ It is rarely worth the effort to change them at the risk of breaking existing co
 
 
 
-**Consider** using a class instead of an interface.
+**Consider** using a class instead of an interface for services and declarables.
```
Comment (gkalpak) [inline aio/content/guide/styleguide.md]: "declarables" sounds very confusing imo.

Comment (Toxicable) [reply aio/content/guide/styleguide.md]: Agreed, I was going to do `Components, Directives and Pipes` but thought it was too long when comparing to other statements

Comment (kapunahelewong) [reply aio/content/guide/styleguide.md]: When I've come across the same thing, Misko has suggested "declarables (components, directives, and pipes)". That's what I've done a couple times in https://github.com/angular/angular/pull/20306.

Comment (Toxicable) [reply aio/content/guide/styleguide.md]: Sounds good to me, fixed

**Review by kapunahelewong — APPROVED** [review]

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**